### DISCUSSION TITLE
[Feature] `before_train_iter` Hook for visualising training batch

### DISCUSSION THREAD

**Discussion by levan92** [open-mmlab/mmengine#855] [General]
### What is the feature?

Are there any existing functionality to visualise (a subset of) training batch? This will be useful if one is using strong/special augmentation or sampling methods, to allow users to know that the training input images are as expected. Should be straightforward to implement as a `before_train_iter` hook? 

### Any other context?

Thank you!

**Comment by HAOCHENYE** [comment]
Hi, you can customize a `VisualizationHook` like [this](https://github.com/open-mmlab/mmdetection/blob/99811076cdf8f6a5a2b0d83419a2cd8ce49ccd73/mmdet/engine/hooks/visualization_hook.py#L18) and implement the `before_train_iter` method to visualize the training data. 

However, visualizing the training data during training will consume a lot of time, we suggest using this script like [this](https://github.com/open-mmlab/mmdetection/blob/master/tools/misc/browse_dataset.py) to check the augmentation result.

**Comment by levan92** [comment]
Thanks for your quick response and the references. You're right about that, probably better to have an offline script iterates through data loader (so custom sampling methods can be checked too).
